### PR TITLE
[ELC-52] Java API updates.

### PR DESCRIPTION
**Pull request by aliakseiTraihel**

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/etoolbox-link-inspector/components/gridConfig/gridConfig.html [hunk 1]
```diff
@@ -9,15 +9,18 @@
         <img itemprop="thumbnail" width="192" src="${model.thumbnail}" alt="">
     </td>
 
-    <td class="foundation-collection-item-title" is="coral-table-cell">
+    <td class="foundation-collection-item-title collection-item_link_type" is="coral-table-cell">
```
Comment (smiakchilo) [inline ui.apps/src/main/content/jcr_root/apps/etoolbox-link-inspector/components/gridConfig/gridConfig.html:12]: Is `collection-item_link_type` the class name that you created with your own hand? If so, you might want to read https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/ about the best practices of CSS naming

**Review by liubou-masiuk — APPROVED** [review]

File: core/src/main/java/com/exadel/etoolbox/linkinspector/core/models/ui/PaginationModel.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        adaptables = SlingHttpServletRequest.class,
+        defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL
+)
+public class PaginationModel {
+
+    private static final String REQUEST_PARAMETER_PAGE = "page";
+    private static final String REQUEST_PARAMETER_TYPE = "type";
+    private static final String REQUEST_PARAMETER_SUBSTRING = "substring";
+    private static final int DEFAULT_PAGE_NUMBER = 1;
+    private static final int DEFAULT_PAGE_SIZE = 500;
```
[the remark this answers is not in the document]
Comment (aliakseiTraihel) [reply core/src/main/java/com/exadel/etoolbox/linkinspector/core/models/ui/PaginationModel.java:33]: https://github.com/exadel-inc/etoolbox-link-inspector/issues/54

File: core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/cache/impl/GridResourcesCacheImpl.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Implements {@link GridResourcesCache} interface to provide data caching functionality.
+ */
+@Component(service = GridResourcesCache.class)
+public class GridResourcesCacheImpl implements GridResourcesCache {
+
+    private static final String ELC_BROKEN_LINKS_CACHE = "elc-broken-links";
```
[the remark this answers is not in the document]
Comment (aliakseiTraihel) [reply core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/cache/impl/GridResourcesCacheImpl.java:20]: etoolbox link checker

File: core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/cache/GridResourcesCache.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.exadel.etoolbox.linkinspector.core.services.cache;
+
+import com.exadel.etoolbox.linkinspector.core.services.data.models.GridResource;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public interface GridResourcesCache {
```
[the remark this answers is not in the document]
Comment (aliakseiTraihel) [reply core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/cache/GridResourcesCache.java:8]: your words make sense too

**Pull request merged**